scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -2243,6 +2244,20 @@ s32 BPF_STRUCT_OPS(layered_init_task, struct task_struct *p,
 		scx_bpf_error("task_ctx allocation failure");
 		return -ENOMEM;
 	}
+	const struct cpumask * all_mask = cast_mask(all_cpumask);
+	if (!all_mask)
+		return -EINVAL;
+
+	if (!(cpumask = bpf_cpumask_create()))
+		return -ENOMEM;
+
+	bpf_cpumask_and(cpumask, all_mask, p->cpus_ptr);
```

Comment:
Yeah, this would work too but you'd also want to update the mask on `ops.set_cpumask()`.